Source/ForecastDeskbarView.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017 Benjamin Amos
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+
+#include <Bitmap.h>
+#include <StringView.h>
+#include <String.h>
+#include <MessageRunner.h>
+#include <Looper.h>
+#include <Roster.h>
+
+#include "ForecastDeskbarView.h"
+#include "ForecastView.h"
+#include "App.h"
+#include "Util.h"
+
+const uint32 kUpdateForecastMessage = 'Updt';
+const float kToolTipDelay = 1000000; /*1000000ms = 1s*/
+
+ForecastDeskbarView::ForecastDeskbarView(BRect viewSize, ForecastView* forecastView)
+	:	BView(viewSize, "ForecastDeskbarView", B_FOLLOW_ALL, B_WILL_DRAW | B_NAVIGABLE | B_PULSE_NEEDED)
+{
+	fForecastView = forecastView;
+}
+
+ForecastDeskbarView::ForecastDeskbarView(BMessage* archive)
+	:	BView(BRect(0, 0, 15, 15), "ForecastDeskbarView", B_FOLLOW_ALL, B_WILL_DRAW)
+{
+	fForecastView = NULL;
+}
+
+ForecastDeskbarView::~ForecastDeskbarView()
+{
+	delete fMessageRunner;
+}
+
+void
+ForecastDeskbarView::AttachedToWindow()
+{
+	fMessageRunner = new BMessageRunner(BMessenger(this), new BMessage(kUpdateForecastMessage), kToolTipDelay, -1);
+	fForecastView->AttachedToWindow();
+	fForecastView->Reload(true);
+
+	AddChild(fForecastView);
+}
+
+status_t
+ForecastDeskbarView::Archive(BMessage* into, bool deep=true) const
+{
+	BView::Archive(into, deep);
```

Comment:
If there is nothing more to archive, you do not need to override the Archive method at all.